### PR TITLE
TP-652 Find and view quotas

### PR DESCRIPTION
**Pull request by DIT-CJWaszczuk**
## Summary 
Added quotas list page with search and filter 
## Screenshots
image

### CODE REVIEW

File: quotas/filters.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        choices=validators.QuotaCategory.choices,
+    )
+
+    clear_url = reverse_lazy("quota-ui-list")
+
+    def filter_origins(self, queryset, name, value):
+        if value:
+            value_array = value.split()
+            return queryset.filter(origins__area_id__in=value_array)
+        return queryset
```
Comment (simonwo) [inline quotas/filters.py:47]: It feels like we might be wanting to use this in multiple places. Maybe we could do with subclassing `Filter` and providing this as a more general component.

Comment (DIT-CJWaszczuk) [reply quotas/filters.py]: I've made `MultiValueCharFilter`, which should be reusable wherever needed now.

File: quotas/filters.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    order_number = CharFilter(
+        label="Order number",
+        field_name="order_number",
+    )
+
+    origin = MultiValueCharFilter(
+        label="Origin(s)",
+        field_name="origins__area_id",
+        method="filter_origins",
```
Comment (simonwo) [inline quotas/filters.py:23]: Do we still need `method` for now?

**Review by simonwo — APPROVED** [review]

File: quotas/filters.py (hunk 2 above)
Comment (DIT-CJWaszczuk) [reply quotas/filters.py]: Nope, removed

File: common/filters.py [hunk 3]
```diff
@@ -46,6 +47,17 @@ def field_to_layout(field_name, field):
     return field_name
 
 
+class MultiValueCharFilter(CharFilter):
+    def filter(self, qs, value):
+        if value in EMPTY_VALUES:
+            return qs
+        if self.distinct:
+            qs = qs.distinct()
+        lookup = "%s__%s" % (self.field_name, "in")
```
Comment (andyhd) [inline common/filters.py:56]: The `%` operator should be treated as deprecated. Use `f"{self.field_name}__in"` instead.

File: common/filters.py [hunk 4]
```diff
@@ -46,6 +47,17 @@ def field_to_layout(field_name, field):
...
+        qs = self.get_method(qs)(**{lookup: value.split()})
+        return qs
```
Comment (andyhd) [inline common/filters.py:58]: Just `return self.get_method...`

File: common/static/common/scss/_table.scss [hunk 5]
```diff
@@ -7,4 +7,11 @@
       font-size: inherit;
     }
   }
+  .govuk-table-list {
+    @extend .govuk-list;
```
Comment (andyhd) [inline common/static/common/scss/_table.scss:11]: This is currently giving me
```
ERROR in ./common/static/common/scss/application.scss
Module build failed (from ./node_modules/mini-css-extract-plugin/dist/loader.js):
ModuleBuildError: Module build failed (from ./node_modules/sass-loader/dist/cjs.js):
SassError: From line 30, column 3 of node_modules/govuk-frontend/govuk/core/_lists.scss: 
   ╷
30 │   .govuk-list {
   │   ^^^^^^^^^^^^
   ╵
You may not @extend selectors across media queries.
   ╷
11 │     @extend .govuk-list;
   │     ^^^^^^^^^^^^^^^^^^^
   ╵
```

**Pull request merged**